### PR TITLE
[r/docs] Update README.md to current build setup

### LINKED ISSUE #1333: [r] Update dev notes in `apis/r/README.md`
As of #1301 and #1302, the developer workflow has changed. This needs to be documented in `apis/r/README.md`.

* Source of truth is `.github/workflows/r-ci.yml` -- let's have the readme point to that
* Additional "color" in the README including "Access to more test data ..." should remain there
* `cleanup` doesn't exist anymore
* `copy_source.sh` doesn't exist anymore
* `scripts/bld --prefix=/usr/local` and `sudo ldconfig` (the latter for Linux) need to be there
* Additional "color" in the README including `R CMD build ...` and `R CMD INSTALL ...` are, of course, obvious to R experts -- but in the spirit of inclusivity and polyglottalism we should retain this helpful information in the readme

See also #1334.

### PR DESCRIPTION
**Pull request by eddelbuettel**
Fixes #1333

**Issue and/or context:**

The README in `apis/r` had drifted slightly from current installation steps.

**Changes:**

Small edits to the README.

**Notes for Reviewer:**

https://github.com/single-cell-data/TileDB-SOMA/issues/1333
[SC 29381](https://app.shortcut.com/tiledb-inc/story/29381/update-readme-md-for-apis-r)

### CODE REVIEW

**Review by johnkerl — APPROVED** [review]
LGTM

Eager to hear from @mojaveazure and @mlin as well, as these notes are primarily for developers & that's us :)

**Review by aaronwolen — APPROVED** [review]
This is really helpful. I made a couple of minor comments/suggestions.

**Review by aaronwolen — APPROVED** [review]
This is really helpful. I made a couple of minor comments/suggestions.

**Review by aaronwolen — APPROVED** [review]
This is really helpful. I made a couple of minor comments/suggestions.

**Review by mojaveazure — APPROVED** [review]

File: apis/r/README.md [hunk 1]
```diff
@@ -1,46 +1,47 @@
 # Overview
 
-This is a R implementation of the [Unified Single-cell Data Model](https://github.com/single-cell-data/SOMA).
+This is the R implementation of the [Unified Single-Cell Data Model](https://github.com/single-cell-data/SOMA).
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:3]: Nice, adopted.

File: apis/r/README.md [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
-* In addition, the R package also depends on the [`libtiledbsoma` library](https://github.com/single-cell-data/TileDB-SOMA/tree/main/libtiledbsoma) -- which gets installed with the package as
-  described in the next section.
+* Source installation requires the [`tiledb` R package](https://github.com/TileDB-Inc/TileDB-R) -- which in turn depends on either a local installation of [TileDB Core library](https://github.com/TileDB-Inc/TileDB) or the provided build artifacts.
+* In general, source installation of TileDB Core and its packages requires `cmake` and `git` to be installed; these are common tools each operating system provides readily.
+* All other R package dependencies are listed in the [DESCRIPTION](https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/r/DESCRIPTION) file and can be installed via _e.g._
+  `remotes::install_deps(".", TRUE)`. In order to build vignettes, `knitr` and `rmarkdown` are required, as is `testthat` for testing. If `testthat` is invoked directly then `pkgbuild` is also needed.
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:28]: Nice catch.  That opens a new can of works / venue for improvement.

I usually use the default (`depedencies=`NA`) which excludes `Suggests:`.  If we set `TRUE` (a good idea) we can also remove several later sentence on `testthat`, `knitr` and all that jazz.  I would be in favor of that: explain here that `dependencies=TRUE` takes care of all Suggests as well.

Comments @johnkerl @aaronwolen @mojaveazure ?

File: apis/r/README.md [hunk 3]
```diff
@@ -1,46 +1,47 @@
...
-* Check and test the package from the tarball: `R CMD check --no-vignettes --no-manual tiledbsoma_*.tar.gz`
-  * For quicker iteration, run `Rscript -e 'testthat::test_local("tests/testthat")'`
-* Install the package from the tarball: `R CMD INSTALL tiledbsoma_*.tar.gz`
-
-Once installed successfully, the package sources can be edited and re-installed.
+* Clone this repository: `git clone https://github.com/single-cell-data/TileDB-SOMA.git`.
+* Change into the R API package directory: `cd TileDB-SOMA/apis/r`.
+* Optionally, clean the files in the repo: `./cleanup` (this is not needed the first time).
+* Optionally, add on optional package with more more test data: `install.packages("pbmc3k.tiledb",  repos="https://ghrr.github.io/drat")`.
+* If you have edited any `src/*.cpp` files and changed any function signaturess, then a run `Rscript -e 'Rcpp::compileAttributes()'` will update the `Rcpp`-generated glue code.
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:38]: Thanks, corrected.

File: apis/r/README.md [hunk 4]
```diff
@@ -1,46 +1,47 @@
...
 
 ## Using R-universe
 
-This code is hosted at R-universe, so you can do
+This package is hosted at R-universe so you can do
 
 ```shell
-$ install.packages('tiledbsoma', repos = c('https://tiledb-inc.r-universe.dev', 'https://cloud.r-project.org'))
+> install.packages('tiledbsoma', repos = c('https://tiledb-inc.r-universe.dev',
+                                           'https://cloud.r-project.org'))
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:16]: I had this discussion with many people over the years and I think the prompt is good and should stay to indicate that this is an R command-line.  The code as rendered can still be marked by mouse and copied.

File: apis/r/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:28]: Upon reflection went the other way and expanded instead.

File: apis/r/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (aaronwolen) [reply apis/r/README.md:16]: But you can't easily take advantage of GitHub's handy copy shortcut:

Screenshot 2023-05-25 at 11 30 07 AM

[the remark this answers is not in the document]
Comment (aaronwolen) [reply apis/r/README.md:16]: We should be consistent across the various readmes within the repo and for now we're using the prompt in other places as well. I personally feel it adds an unnecessary speed bump but we can address in a separate PR to let others weigh in.

[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:16]: Indeed we should be in many / most aspects.  And the README for the `tiledb` package has a `>` too.

[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:16]: Let me know if you are ok to proceed.

[the remark this answers is not in the document]
Comment (aaronwolen) [reply apis/r/README.md:16]: Yeap, totally okay to proceed.  This will definitely help clear up some of the confusion around building the R package.

**Pull request merged**